src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.DiagnosticProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CodeFixes
+{
+    /// <summary>
+    /// Context for "Fix all occurrences" code fixes provided by an <see cref="FixAllProvider"/>.
+    /// </summary>
+    public partial class FixAllContext
+    {
+        /// <summary>
+        /// Diagnostic provider to fetch document/project diagnostics to fix in a <see cref="FixAllContext"/>.
+        /// </summary>
+        public abstract class DiagnosticProvider
```

Comment:
This is the new type @heejaechang proposed. I have made it a nested type in FixAllContext to make the connection with FixAll very explicit.